submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,313 @@
+         project: ns1
+     podSelector:
+         matchLabels:
+           role: db
+```
+
+5.In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
+  to use a unique globalIP both as ingress and egressIP for cross-cluster communication.
+
+  To achieve this, user can create a Headless Service (backed by a Deployment or StatefulSets) and
+  Globalnet will ensure that all the backend Pods that belong to the Headless Service are assigned
+  a globalIP. User is also expected **not** to create any GlobalnetEgressIP object which selects the
+  backendPods that match the Headless Service.
+
+![HeadlessService](./images/globalnet-headless-svc.png)
+
+  In the above example, each of the http pods will get its own globalIP which can be used for
+  both ingress and egress communication.
+
+  However, if the user wants egress traffic from the http Pods to carry a unique globalIP,
+  its possible by explicitly requesting a GlobalnetEgressIP with PodSelectors matching Headless
+  Service Pods as shown below.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalnetEgressIP
+   metadata:
+     name: http-pods
+     namespace: myproject
+   spec:
+     namespaceSelector:
+       matchLabels:
+         project: ns1
+     podSelector:
+       matchLabels:
+         app: http
+   status:
+     globalIPs:
+      - 169.254.0.100
+```
+
+![Headless Service with EgressIP](./images/globalnet-svc.png)
+
+## Alternatives
+
+It is possible to support Pod to Pod communication even with Globalnet. However, to support this, it
+requires programming additional IPtable rules on the nodes which will have an impact on the performance
+of the solution. This proposal provides some limited Pod to Pod connectivity via the Headless Services.
+If user is interested in Pod to Pod communication for all the Pods, it is recommended to install
+Clusters with non-overlapping CIDRs and use Vanilla Submariner instead of Globalnet.
+
+## Work items
+
+1. Support creation of GlobalnetEgressIP CRD and the associated Cluster Roles.
+2. Provide configuration parameter either via ConfigMap or Env variable to specify
+   the number of default GlobalIPs that have to be allocated at cluster level.
+3. Support Globalnet EgressIPs at the cluster level and program necessary egress rules.
+4. Support Globalnet EgressIPs at namespace level which takes precedence over
+   globalIPs at cluster level.
+5. Support Globalnet EgressIPs to Pods (or set of Pods) which has the highest precedence.
+6. Annotate only exported Services instead of annotating all Services.
+7. Support Headless Services and program ingress/egress rules for the backend Pods.
+8. Avoid dependency on the iptable chains programmed by IPtables kube-proxy driver.
+9. Implement necessary unit tests for each of the use-cases.
+10. Implement necessary e2e tests to validate various use-cases.
+11. Update documentation.
+12. Support upgrade from 0.8 release to the newer implementation.
```

Comment:
How will this work? Didn't you mention the change is not backwards compatible?

Reply:
Done, I'll update the action item.